dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/CompanyDataController.kt

```diff
@@ -47,19 +54,49 @@ class CompanyDataController(
         )
         return ResponseEntity.ok(
             companyManager.searchCompaniesAndGetApiModel(
-                CompanySearchFilter(
-                    searchString ?: "",
-                    onlyCompanyNames,
-                    dataTypes ?: setOf(),
-                    countryCodes ?: setOf(),
-                    sectors ?: setOf(),
-                    onlyCurrentUserAsUploader,
+                StoredCompanySearchFilter(
+                    searchString = searchString ?: "",
+                    nameOnlyFilter = onlyCompanyNames,
+                    dataTypeFilter = dataTypes.map { it.name },
+                    countryCodeFilter = countryCodes?.toList() ?: listOf(),
+                    sectorFilter = sectors?.toList() ?: listOf(),
+                    uploaderIdFilter = getUploaderIdFilter(onlyCurrentUserAsUploader),
```

Comment:
Generell zu den Filtern:  Wieso sind wir auf Listen umgeswitcht?   Sets sind doch ganz nett, weil dann der duplicate-Fall direkt ausgeschlossen ist, oder?

Reply:
Wir sind nur an der Stelle geswitcht, weil wir hier jetzt den StoredCompanySearchFilter benutzen. Vorher war das halt in `companyManager.searchCompaniesAndGetApiModel`. Duplicates sind ausgeschlossen, da es ja ursprünglich Sets waren. Warum wir keine Sets reingeben? Weiß ich nicht. Vielleicht mag die Query das nicht.

Reply:
Ok dann lassen wir es so mit Listen.    Dürfte der SQL query am Ende egal sein, so wie ich das sehe.